modules/camera/camera_win.cpp

```diff
@@ -64,35 +414,293 @@ CameraFeedWindows::~CameraFeedWindows() {
 	};
 
 	///@TODO free up anything used by this
+	DeleteCriticalSection(&m_critsec);
 };
 
 bool CameraFeedWindows::activate_feed() {
+	HRESULT hr = S_OK;
+	IMFMediaSource *pSource = NULL;
+	IMFAttributes *pAttributes = NULL;
+	IMFMediaType *pType = NULL;
+	UINT32 p_type_str_len = 0;
+	_Post_ _Notnull_ LPWSTR p_type_str = NULL;
+
+	GUID subtype = { 0 };
+	MFRatio PAR = { 0 };
+
 	///@TODO this should activate our camera and start the process of capturing frames
+	printf("activate_feed\n");
 
-	return true;
-};
+	EnterCriticalSection(&m_critsec);
+
+	hr = CloseDevice();
+
+	// Create the media source for the device.
+	if (SUCCEEDED(hr)) {
+		hr = pActivate->ActivateObject(
+				__uuidof(IMFMediaSource),
+				(void **)&pSource);
+
+		if (FAILED(hr)) {
+			printf("ActivateObject Failed\n");
+		}
+	}
+
+	// Get the symbolic link
+	if (SUCCEEDED(hr)) {
+		hr = pActivate->GetAllocatedString(
+				MF_DEVSOURCE_ATTRIBUTE_SOURCE_TYPE_VIDCAP_SYMBOLIC_LINK,
+				&m_pwszSymbolicLink,
+				&m_cchSymbolicLink);
+		if (FAILED(hr)) {
+			printf("Getting Symbolic Link Failed\n");
+		}
+	}
+
+	// Create an attribute store to hold initialization settings.
+	if (SUCCEEDED(hr)) {
+		hr = MFCreateAttributes(&pAttributes, 2);
+		if (FAILED(hr)) {
+			printf("Attribute creation failed\n");
+		}
+	}
+
+	if (SUCCEEDED(hr)) {
+		hr = pAttributes->SetUINT32(MF_READWRITE_DISABLE_CONVERTERS, TRUE);
+		if (FAILED(hr)) {
+			printf("Set attribute (disable converter) failed\n");
+		}
+	}
+
+	//trying to stop shutdown() so that we can reactivate feed
+	if (SUCCEEDED(hr)) {
+		hr = pAttributes->SetUINT32(MF_SOURCE_READER_DISCONNECT_MEDIASOURCE_ON_SHUTDOWN, TRUE);
+		if (FAILED(hr)) {
+			printf("Set attribute (disable shutdown on disconnect) failed\n");
```

Comment:
Nitpicking but it is better to use Godots internal print methods so the messages get locked to the log in the editor. You can also switch between what is considered info and what is actually an error.

Reply:
ok. i am going to do some tidying up as well.